rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -140,7 +143,8 @@ class Player : public rclcpp::Node
   static constexpr double read_ahead_lower_bound_percentage_ = 0.9;
   static const std::chrono::milliseconds queue_read_wait_period_;
 
-  std::unique_ptr<rosbag2_cpp::Reader> reader_;
+  std::mutex reader_mutex_;
+  std::unique_ptr<rosbag2_cpp::Reader> reader_ RCPPUTILS_TSA_GUARDED_BY(reader_mutex_);
```

Comment:
@emersonknapp This is what I was talking on today's meeting. Just adding a few member variables causing memory corruption in constructor.  

TEST_F(RosBag2PlayTestFixture, player_can_resume_after_play_next) fails with memory corruption in Player constructor.

Reply:
I'm also tracking down a memory corruption in https://github.com/ros-simulation/gazebo_ros_pkgs/issues/1296 .  I'm wondering if they are related.  If you create the node with `NodeOptions.use_clock_thread(false)`, does that make any difference?

Reply:
@clalancette Nop. The corruption still there even with `NodeOptions.use_clock_thread(false)`

Reply:
> @clalancette Nop. The corruption still there even with `NodeOptions.use_clock_thread(false)`.
> To be precise I observe exact the same issue with exact the same report from Valgrind.

All right, thanks for giving it a shot.  Seems like a different problem then.